### PR TITLE
feat(suite): implement the anonymisation progress selector

### LINKED ISSUE #7108: Change logic of progress calculation of CoinJoin progress bar
**Current behaviour:**
`signed rounds` / `confirmed rounds`

**Proposed improvement:**
$AL$ = `anonymity level` - 1
$DAL$ = `desired anonymity level` -1

$$progress = {\sum(amount_i* min(AL_i,DAL))\over\sum amount_i * DAL}$$

If $DAL = 0$ then $progress = 1$

### PR DESCRIPTION
**Pull request by dahaca**
## Description

Re-implemented anonymisation progress calculation

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/7108

### CODE REVIEW

**Comment by hynek-jina** [comment]
Hi @dahaca,
the calculation doesn't seem correct..

Visible progress is circa 23%
Snímek obrazovky 2023-01-06 v 15 28 08

With desired anonymity level 5
Snímek obrazovky 2023-01-06 v 15 28 13

And following UTXO (and reached anonymity)
Snímek obrazovky 2023-01-06 v 15 28 41

**The progress should be much lower..**

**Comment by dahaca** [comment]
> The progress should be much lower..

@hynek-jina  My apologies, forgot to exclude the ones with anonymity **1**, will push a fix soon.

File: packages/suite/src/utils/wallet/coinjoinUtils.ts [hunk 1]
```diff
@@ -61,6 +61,45 @@ export const breakdownCoinjoinBalance = ({
...
+        return 0;
+    }
+
+    let weightedCurrentAnonymity = new BigNumber(0);
+    let weightedTargetAnonymity = new BigNumber(0);
+
+    utxos?.forEach(({ address, amount }) => {
+        const bigAmount = new BigNumber(amount);
+
+        const currentAnonymity = anonymitySet[address] || 0;
```
Comment (matejkriz) [inline packages/suite/src/utils/wallet/coinjoinUtils.ts]: 0 would go to -1 on L89
```suggestion
        const currentAnonymity = anonymitySet[address] || 1;
```

File: packages/suite/src/utils/wallet/coinjoinUtils.ts [hunk 2]
```diff
@@ -61,6 +61,45 @@ export const breakdownCoinjoinBalance = ({
...
+
+        weightedCurrentAnonymity = weightedCurrentAnonymity.plus(
+            bigAmount.times(Math.min(currentAnonymity, targetAnonymity) - 1),
+        );
+        weightedTargetAnonymity = weightedTargetAnonymity.plus(
+            bigAmount.times(targetAnonymity - 1),
```
Comment (matejkriz) [inline packages/suite/src/utils/wallet/coinjoinUtils.ts:100]: I know it should not be reachable in our current flow, but still would be nice to handle the case when `targetAnonymity === 1`

```javascript
if(targetAnonymity === 1) {
  return 100;
}
```

File: packages/suite/src/utils/wallet/__fixtures__/coinjoinUtils.ts [hunk 3]
```diff
@@ -2,6 +2,79 @@ import {
...
+            ...baseUtxo,
+        },
+        {
+            address: 'three',
+            ...baseUtxo,
+        },
+    ],
+};
+
+export const calculateProgressParams: Parameters<
```
Comment (matejkriz) [inline packages/suite/src/utils/wallet/__fixtures__/coinjoinUtils.ts]: would be nice to add more critical cases like no utxos, just one utxo, everything above targetAnonymity...

File: packages/suite/src/utils/wallet/coinjoinUtils.ts (hunk 1 above)
Comment (dahaca) [reply packages/suite/src/utils/wallet/coinjoinUtils.ts]: f56eaa083acc69a8652848b738da50547c3beaa6

File: packages/suite/src/utils/wallet/coinjoinUtils.ts (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/utils/wallet/coinjoinUtils.ts:100]: f56eaa083acc69a8652848b738da50547c3beaa6

File: packages/suite/src/utils/wallet/__fixtures__/coinjoinUtils.ts (hunk 3 above)
Comment (dahaca) [reply packages/suite/src/utils/wallet/__fixtures__/coinjoinUtils.ts]: f56eaa083acc69a8652848b738da50547c3beaa6

**Review by matejkriz — APPROVED** [review]
👍

**Pull request merged**